src/pattern-library/components/patterns/feedback/DialogPage.tsx

```diff
@@ -0,0 +1,122 @@
+import { useState, useRef } from 'preact/hooks';
+
+import { Dialog } from '../../../../components/feedback';
+import type { DialogProps } from '../../../../components/feedback/Dialog';
+import {
+  ArrowRightIcon,
+  Button,
+  EditIcon,
+  IconButton,
+  Input,
+  InputGroup,
+} from '../../../../next';
+import Library from '../../Library';
+
+function DialogButtons() {
+  return (
+    <>
+      <Button key="maybe" onClick={() => alert('You chose maybe')}>
+        Maybe
+      </Button>
+      <Button
+        key="yep"
+        onClick={() => alert('You chose yep')}
+        variant="primary"
+      >
+        Do it!
+      </Button>
+    </>
+  );
+}
+
+type Dialog_Props = DialogProps & {
+  /** Pattern-wrapping prop. Not visible in source view */
+  _nonCloseable?: boolean;
+};
+
+/**
+ * Wrap the Dialog component with some state management to make reuse in
+ * multiple examples plausible and convenient.
+ */
+function Dialog_({
+  buttons,
+  _nonCloseable,
+  children,
+  ...dialogProps
+}: Dialog_Props) {
+  const [dialogOpen, setDialogOpen] = useState(false);
+  const closeDialog = () => setDialogOpen(false);
+
+  const closeHandler = _nonCloseable ? undefined : closeDialog;
+  const forwardedButtons = buttons ? (
+    buttons
+  ) : (
+    <Button onClick={closeDialog}>Escape!</Button>
+  );
+
+  if (!dialogOpen) {
+    return (
+      <Button onClick={() => setDialogOpen(!dialogOpen)} variant="primary">
+        Show dialog
+      </Button>
+    );
+  } else {
+    return (
+      <Dialog
+        buttons={forwardedButtons}
+        {...dialogProps}
+        onClose={closeHandler}
+      >
+        {children}
+      </Dialog>
+    );
+  }
```

Comment:
Just picky comment, but you can avoid the `else` block, as the `if` block is returning:

```ts
if (!dialogOpen) {
  return (
    <Button ... />
  );
}

return (
  <Dialog ... />
);
```

Reply:
Fair enough! I'll rectify in the next pass.